docs/Porting_Guide.md

```diff
@@ -47,3 +48,86 @@ Understanding the TizenRT code structure and adhering to the structure helps in
+
+		 -----------------------------------------------------------------------------------------------
+		|    CRC32   | Header size | Binary Version |  Binary size | Secure Header size | Secure Header |
+		|  (4 bytes) |  (2 bytes)  |   (4 bytes)    |   (4 bytes)  |     (2 bytes)      |   (# bytes)   |
+		 -----------------------------------------------------------------------------------------------
+
+These binary header components are as follows:
+
+#### 1. CRC32
+This is the CRC32 value for all header and actual TizenRT binary. It is checksum value for the entire binary from Header size to the end of binary. It is of type uint32_t and its size is 4 bytes.
+
+#### 2. Header size
+This is the header total data size excluding CRC32. It is of type uint16_t and its size is 2 bytes.
+
+#### 3. Binary version
+This is the version of the binary in "YYMMDD" format (CONFIG_BINARY_VERSION). It is of type uint32_t and its size is 4 bytes.
+
+#### 4. Binary size
+This is the actual TizenRT binary size excluding header size of the binary. It is of type uint32_t and its size is 4 bytes.
+
+#### 5. Secure header size
+This is the size of the secure header and it depends on the board. It is of type uint16_t and its size is 2 bytes.
+
+#### 6. Secure header
+This is the secure header signature. It is of type char and its size may vary depending upon the securing method.
+Note: The size of the secure header and the entire binary may vary depending on the secure method in the future.
+
+## Partitions
+
+								   Binary 1			Binary 2
+									|			    |
+								 ----------------------------	 ---------------
+								| Header|    TizenRT Binary  |	| H |		|
+	 --------------------------------------------------------------------------------------------------------------------
+	|  km0 boot  |  km4 boot  |  reserved  | secure storage |	    kernel		|	kernel		| .. |
+	 --------------------------------------------------------------------------------------------------------------------
+	|     16     |	    8     |     16     |       472      |	     1792		|	 1792		| .. |
+	 --------------------------------------------------------------------------------------------------------------------
+
+			Figure: Flash partition information of flash for Realtek board
+
+#### 1. Bootloader
+The bootloader section is placed at the start of the flash. In the above example realtek comprises of two bootloaders namely *km0 boot* & *km4 boot* of sizes 16 and 8 bytes respectively.
+
+#### 2. Reserved
+The reserved partition is next to bootloader. Its size is 16 bytes.
+
+#### 3. Secure Storage
+The secure storage partitions placed next to the reserved partition in the flash. Its 472 bytes in size.
+
+#### 4. Binary
+The header is placed at the start of the binary. Post the binary header lies the TizenRT kernel binary. In the above example realtek comprises of two kernel binaries of sizes 1792 bytes respectively.
+
+### To add binary header, user should take the following steps:
```

Comment:
```
## How to add TizenRT Header
User should define MAKE_SAMSUNG_HEADER for binary header support in *build/configs/<board>/<name>/Make.defs as below:
``
KERNEL_BINARY_PATH='xxxx.bin'
SECURE_HEADER_SIZE=xx

define MAKE_SAMSUNG_HEADER
  $(TOPDIR)/../os/tools/mksamsungheader.py ${KERNEL_BINARY_PATH} kernel ${SECURE_HEADER_SIZE}
endef
``

where,
KERNEL_BINARY_PATH is the path where kernel binary is generated and stored.
For example, in case of realtek it is generated here:
``
build/output/bin/km0_km4_image2.bin
``
Similarly in case of imxrt, the kernel binary is generated here:
``
build/output/bin/tinyara.bin
``
SECURE_HEADER_SIZE is the size of the secure header.
```

Reply:
updated